cpp_test_suite/new_tests/cxx_dserver_misc.cpp

```diff
@@ -255,6 +255,50 @@ cout << "str = " << str << endl;
         TS_ASSERT_EQUALS(0, callback.num_of_error_events);
     }
 
+/*
+ * Test to check that client still receives periodic archive events after
+ * polling is stopped and started again for particular device. Such scenario
+ * used to fail as described in #675.
+ */
+    void test_archive_periodic_events_after_polling_restart()
+    {
+        EventCallback callback{};
+
+        std::string attribute_name = "PollLong_attr";
+
+        auto config = device1->get_attribute_config(attribute_name);
+        config.events.arch_event.archive_period = std::to_string(1500);
+        AttributeInfoListEx config_in = { config };
+        TS_ASSERT_THROWS_NOTHING(device1->set_attribute_config(config_in));
+
+        TS_ASSERT_THROWS_NOTHING(device1->poll_attribute(attribute_name, 1500));
+
+        TS_ASSERT_THROWS_NOTHING(device1->subscribe_event(
+            attribute_name,
+            Tango::ARCHIVE_EVENT,
+            &callback));
+
+        Tango_sleep(2);
+        TS_ASSERT_EQUALS(0, callback.num_of_error_events);
+        TS_ASSERT_EQUALS(2, callback.num_of_all_events);
+
+        TS_ASSERT_THROWS_NOTHING(device1->stop_poll_attribute(attribute_name));
+
+        Tango_sleep(2);
+        TS_ASSERT_EQUALS(1, callback.num_of_error_events);
+        TS_ASSERT_EQUALS(3, callback.num_of_all_events);
+
+        TS_ASSERT_THROWS_NOTHING(device1->poll_attribute(attribute_name, 1500));
+
+        Tango_sleep(2);
+        TS_ASSERT_EQUALS(1, callback.num_of_error_events);
+        TS_ASSERT_EQUALS(4, callback.num_of_all_events);
```

Comment:
It looks like we could get 5 events received in some cases here, depending on the synchronization of the polling thread.
At line 291, the attribute polling is restarted so it is polled every 1.5 s.
At line 293, we sleep 2 seconds.
During these 2 seconds, the polling thread has potentially the time to read twice the attribute, so, to send 2 events, isn't it?
I got 1 test failure like this once when I tried to reproduce it on an Ubuntu 18.04 VM:
`/home/reynald/GIT/mliszcz/cppTango/cpp_test_suite/cxxtest/include/../../new_tests/cxx_dserver_misc.cpp:295: Error: Expected (4 == callback.num_of_all_events), found (4 != 5)`

Reply:
Changed. Now it is polled every 1s and after polling period I wait events for 100ms (so it is not possible for this buffer to overlap with next polling period).